mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/KeyRegistrationHandler.java

```diff
@@ -125,20 +125,25 @@ void stop() {
             .block();
     }
 
-    Registration register(MailboxListener.ReactiveMailboxListener listener, RegistrationKey key) {
+    Mono<Registration> register(MailboxListener.ReactiveMailboxListener listener, RegistrationKey key) {
         LocalListenerRegistry.LocalRegistration registration = localListenerRegistry.addListener(key, listener);
+
+        return registerIfNeeded(key, registration)
+            .thenReturn(new KeyRegistration(() -> {
+                if (registration.unregister().lastListenerRemoved()) {
+                    registrationBinder.unbind(key)
+                        .retryWhen(Retry.backoff(retryBackoff.getMaxRetries(), retryBackoff.getFirstBackoff()).jitter(retryBackoff.getJitterFactor()).scheduler(Schedulers.elastic()))
+                        .block();
+                }
```

Comment:
A `block` in Reactor?

Reply:
Well that is in the unregister call back so it is not directly related to "register" but more "unregister"

I agree to push that work forward and reactify `unregister`.

(And if any :fr: people have some spare time, I would be happy to accept help on that)

Reply:
While reactifying unregistration might be beneficial, I believe the true important thing here is a ` .subscribeOn(Schedulers.elastic())` call (that I contributed here). unregister is not called by reactor code thus the criticity is low and we might choose to delay this refactoring.